compiler/rustc_trait_selection/src/traits/error_reporting/mod.rs

```diff
@@ -691,6 +691,25 @@ impl<'tcx> TypeErrCtxtExt<'tcx> for TypeErrCtxt<'_, 'tcx> {
                         {
                             return;
                         }
+                        if let ObligationCauseCode::FunctionArgumentObligation {
+                            arg_hir_id,
+                            ..
+                        } = obligation.cause.code()
+                            && let Some(Node::Expr(arg)) = self.tcx.hir().find(*arg_hir_id)
```

Comment:
I think I'd like to see the stuff inside the `if let` here put into a function, just to keep `report_selection_error` a bit simpler.

Reply:
@estebank does this comment needs to be addressed?

Apart from this I think this PR can be r+'ed by @davidtwco (as per [comment](https://github.com/rust-lang/rust/pull/113183#pullrequestreview-1593299211))?

Reply:
This was already done